providers/stores/file.js

```diff
@@ -21,8 +24,18 @@ class FileStore extends AbstractStore {
     try {
       const paths = await recursive(this._toStoragePathFromCoordinates(coordinates), ['.DS_Store'])
       const list = new Set()
-      paths.forEach(entry => {
-        const value = this._getEntry(entry, type)
+      const files = await Promise.all(
+        paths.map(path => {
+          return new Promise((resolve, reject) =>
+            fs.readFile(path, (error, data) => (error ? reject(error) : resolve({ path, contents: JSON.parse(data) })))
+          )
+        })
+      )
+      files.forEach(file => {
```

Comment:
Interesting, here I would have combined these two loops and just done something like

```javascript
return Promise.all(paths.map(path => {
  if (type !== entity) return this._getEntry(path, type).toString()
  return new Promise((resolve, reject) =>
    fs.readFile(path, (error, data) => {
        if (error) return reject(error)
        const contents = JSON.parse(data)
        resolve(this._getEntry(contents._metadata.coordinates , type)).toString()
}).filter(e => e))
```

and uniquified at the end.

Reply:
Right! Only read the file if we have to. Will update this

Reply:
since we're going to preserve case if `type == entity || type == result` I ended up always reading the file

Reply:
follow up on this to combine into a single loop